astroid/inference.py

```diff
@@ -197,15 +201,19 @@ def _higher_function_scope(node):
         otherwise an instance of :class:`astroid.nodes.scoped_nodes.Function`,
         which encloses the given node.
     """
-    current = node
+    current: nodes.NodeNG = node
     while current.parent and not isinstance(current.parent, nodes.FunctionDef):
         current = current.parent
     if current and current.parent:
-        return current.parent
+        return current.parent  # type: ignore[return-value]
     return None
 
 
-def infer_name(self, context=None):
+def infer_name(
+    self: nodes.Name | nodes.AssignName,
+    context: InferenceContext | None = None,
+    **kwargs: Any,
+) -> Generator[InferenceResult, None, None]:
     """infer a Name: use name lookup rules"""
     frame, stmts = self.lookup(self.name)
```

Comment:
I believe the return type for `self.lookup` might be wrong. It returns a generator from `scope_lookup`. However, this does return a `NodeNG` as first tuple argument (usually `self`) and not a string.

Reply:
Introduced in https://github.com/PyCQA/astroid/pull/1503/.

Never trust typing in `astroid` docstrings 😄